src/cascadia/ShellExtension/OpenTerminalHere.cpp

```diff
@@ -0,0 +1,206 @@
+        // family name, that means we're one of the packaged Dev build, packaged
+        // Release build, or packaged Preview build.
+        //
+        // If we're the preview or release build, there's no way of knowing if the
+        // `wt.exe` on the %PATH% is us or not. Fortunately, _our_ execution alias
+        // is located in "%LOCALAPPDATA%\Microsoft\WindowsApps\<our package family
+        // name>", _always_, so we can use that to look up the exe easier.
+        try
+        {
+            const auto package{ winrt::Windows::ApplicationModel::Package::Current() };
+            const auto id = package.Id();
+            const std::wstring pfn{ id.FamilyName() };
+            if (!pfn.empty())
+            {
+                const std::filesystem::path windowsAppsPath{ wil::ExpandEnvironmentStringsW<std::wstring>(L"%LOCALAPPDATA%\\Microsoft\\WindowsApps\\") };
+                const std::filesystem::path wtPath = windowsAppsPath / pfn / (IsDevBuild() ? L"wtd.exe" : L"wt.exe");
+                return wtPath;
+            }
+        }
+        CATCH_LOG();
+
+        // If we're here, then we couldn't resolve our exe from the package. This
+        // means we're running unpackaged. We should just use the
+        // WindowsTerminal.exe that's sitting in the directory next to us.
+        try
+        {
+            HMODULE hModule = GetModuleHandle(nullptr);
+            THROW_LAST_ERROR_IF(hModule == nullptr);
+            std::wstring dllPathString;
+            THROW_IF_FAILED(wil::GetModuleFileNameW(hModule, dllPathString));
+            const std::filesystem::path dllPath{ dllPathString };
+            const std::filesystem::path rootDir = dllPath.parent_path();
+            std::filesystem::path wtPath = rootDir / "WindowsTerminal.exe";
+            return wtPath;
+        }
+        CATCH_LOG();
+
+        return L"wt.exe";
+    }();
+    return exePath;
+}
+
+// Method Description:
+// - This method is called when the user activates the context menu item. We'll
+//   launch the Terminal using the current working directory.
+// Arguments:
+// - psiItemArray: a IShellItemArray which contains the item that's selected.
+// Return Value:
+// - S_OK if we successfully attempted to launch the Terminal, otherwise a
+//   failure from an earlier HRESULT.
+HRESULT OpenTerminalHere::Invoke(IShellItemArray* psiItemArray,
+                                 IBindCtx* /*pBindContext*/)
+{
+    DWORD count;
+    psiItemArray->GetCount(&count);
+
+    winrt::com_ptr<IShellItem> psi;
+    RETURN_IF_FAILED(psiItemArray->GetItemAt(0, psi.put()));
+
+    PWSTR pszName;
+    RETURN_IF_FAILED(psi->GetDisplayName(SIGDN_FILESYSPATH, &pszName));
+
+    {
+        wil::unique_process_information _piClient;
+        STARTUPINFOEX siEx{ 0 };
+        siEx.StartupInfo.cb = sizeof(STARTUPINFOEX);
+
+        std::wstring cmdline = fmt::format(L"\"{}\" -d \"{}\"", _getExePath(), pszName);

```

Comment:
if you invoke this in `C:\` it will explode. Want to make it `\"{}\\.\"` (add `\.` to make it a dir?)

Reply:
Okay, I can do that easy